src/Spectre.libStatistics/Math.h

```diff
@@ -380,4 +381,43 @@ std::vector<DataType> build(size_t size, Generator generator)
     return result;
 }
 
+/// <summary>
+/// Calculates differences between adjacent elements of given container.
+/// The function is based upon Matlab function diff():
+/// https://www.mathworks.com/help/matlab/ref/diff.html
+/// but it can handle only one dimensional data.
+/// </summary>
+/// <param name="data">The container.</param>
+/// <param name="order">Order of difference operation (how many times the difference is executed).</param>
+/// <returns>Vector of calculated differences, of size: length_of_data - order.</returns>
+template <class DataType>
+std::vector<typename std::remove_const<DataType>::type> differentiate_unsafe(gsl::span<DataType> data, uint16_t order = 1)
```

Comment:
We'll need to preapre an issue for revising the naming convention in this header file.